gprMax/waveforms.py

```diff
@@ -102,6 +103,12 @@ def calculate_value(self, time, dt):
                 ampvalue = 0
             else:
                 ampvalue = self.uservalues[index]
+        elif self.type == 'userfunc':
+                            
+            if self.userfunc is None:
+                ampvalue = 0
+            else:
+                ampvalue = float(self.userfunc(time))
```

Comment:
userfunc is generated in input_cmds_singleuse.py with a univariate spline, which provides smooth interpolation between all of the data points provided in the input file.

Here is a closeup view of what this looks like in practice:
![image](https://cloud.githubusercontent.com/assets/700223/25006556/5d0cb408-202b-11e7-9bc3-dc646788fb8c.png)

The red dots are the original data points provided by the user input file, while the blue points are from the userfunc sample at a much finer timescale.

If you want to test it yourself:
```
excitationfile = 'waveform_usergauss2.txt'

# Read all waveform values into an array
waveformvalues = np.loadtxt(excitationfile, skiprows=1, dtype=floattype)

timedata = waveformvalues[:, 0]

userfunc = UnivariateSpline(timedata,waveformvalues[:, 1],ext=1,k=2,s=0)

time2 = np.linspace(0,timedata[-1],10000)

plt.plot(time2,userfunc(time2),'b.')
plt.plot(timedata,waveformvalues[:, 1],'ro')
plt.xlim(.45e-9,.55e-9)
plt.ylim(.9,1)
plt.show()
```
Hope this makes sense!